sqld/src/query_analysis.rs

```diff
@@ -95,17 +98,23 @@ impl Statement {
         Self {
             stmt: s.to_string(),
             kind: StmtKind::Write,
+            is_iud: false,
         }
     }
 
     pub fn parse(s: &str) -> impl Iterator<Item = Result<Self>> + '_ {
         fn parse_inner(c: Cmd) -> Result<Statement> {
             let kind =
                 StmtKind::kind(&c).ok_or_else(|| anyhow::anyhow!("unsupported statement"))?;
+            let is_iud = matches!(
+                c,
+                Cmd::Stmt(Stmt::Insert { .. } | Stmt::Update { .. } | Stmt::Delete { .. })
+            );
```

Comment:
It feels like this should be embedded in StmtKind. Maybe it lacks granularity and should be extended

Reply:
Yeah, I am a bit uneasy about this. I didn't want to modify `StmtKind` to avoid interfering with the query analysis code; the only reason to distinguish `INSERT`, `UPDATE` and `DELETE` from other writes like `ALTER TABLE` is to track the `affected_row_count`. But if you think that's ok, maybe I could split `StmtKind::Write` to `StmtKind::WriteRows` (for IUD) and `StmtKind::WriteOther`?

Reply:
let's go with that solution for now, and think about what to do with `StmtKind` later